File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0)return 0;
        sort(nums.begin(),nums.end());
        int ans = 0;
        int temp = 1;
        for(int i=1;i<nums.size();i++){
            if(nums[i]==nums[i-1])continue;
            if(nums[i]==nums[i-1]+1)temp++;
            else{
                ans = max(temp,ans);
                temp = 1;
            }
        }
        ans = max(temp,ans);

        return ans;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int minnum = Integer.MAX_VALUE;
        int ans = 0;

        for(int x : prices){
            ans = Math.max(ans,x-minnum);
            minnum = Math.min(minnum,x);
        }
        return ans;
    }
}
